### PR TITLE
Fix a bug that potentially blocks #1502.

### PR DESCRIPTION
**Pull request by wujingyue**

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: csrc/scheduler/reduction.cpp [hunk 1]
```diff
@@ -1225,6 +1226,12 @@ void scheduleReduction(Fusion* fusion, const ReductionParams& rparams) {
       cached_outputs);
 
   scheduler_utils::promoteProducerMemoryTypes(fusion, cached_inputs);
+
+  // TODO(#1401): We could let segmentation split a partially alias-producing
+  // fusion into an alias-only segment and the rest. This way, the rest of the
+  // fusion (which has fewer expressions) can potentially find a better
+  // scheduler and we need to call markAliases only in NoOpScheduler.
+  markAliases(fusion);
```
Comment (jjsjann123) [inline csrc/scheduler/reduction.cpp:1234]: Am I understanding this correctly that `markAliases` should potentially be called in every scheduler? should we have added it in maybe transpose scheduler as well?

Comment (wujingyue) [reply csrc/scheduler/reduction.cpp:1234]: Yes! I'm adding it incrementally to reduce the blast radius. I was also hoping (as the comment says) segmentation is improved so only NoOpScheduler needs to mark aliases, but I haven't prioritized that yet without a concrete need.

Comment (jjsjann123) [reply csrc/scheduler/reduction.cpp:1234]: Got'ya.

**Review by jjsjann123 — APPROVED** [review]

**Pull request merged**